assets/model_monitoring/components/src/model_data_collector_preprocessor/spark_run.py

```diff
@@ -33,16 +37,33 @@ def handle_data_not_found():
 
     add_tags_func = add_tags_func or add_tags_to_root_run
 
-    file_list = get_file_list(start_datetime, end_datetime, store_url=store_url)
-    if not file_list:
+    try:
+        return _uri_folder_to_spark_df(start_datetime, end_datetime, store_url, soft_delete_enabled=False)
+    except FileNotFoundError:
         handle_data_not_found()
+    except Py4JJavaError as pe:
+        if "This endpoint does not support BlobStorageEvents or SoftDelete" in pe.java_exception.getMessage():
+            blockblob_url = copy_appendblob_to_blockblob(store_url, start_datetime, end_datetime)
+            return _uri_folder_to_spark_df(start_datetime, end_datetime, blockblob_url, soft_delete_enabled=True)
+        else:
+            raise pe
+
+
+def _uri_folder_to_spark_df(start_datetime: datetime, end_datetime: datetime, store_url: StoreUrl,
+                            soft_delete_enabled: bool = False) -> DataFrame:
+    scheme = "azureml" if soft_delete_enabled else "abfs"
+    file_list = get_file_list(start_datetime, end_datetime, store_url=store_url, scheme=scheme)
+    if not file_list:
+        raise FileNotFoundError(f"No data found for the given time window: {start_datetime} to {end_datetime}")
     # print("DEBUG file_list:", file_list)
 
     spark = init_spark()
-    set_data_access_config(spark, store_url=store_url)
+    if not soft_delete_enabled:
+        # need to set credential in spark conf for abfs access
+        set_data_access_config(spark, store_url=store_url)
     df = spark.read.json(file_list)
     if df.rdd.isEmpty():
-        handle_data_not_found()
+        raise FileNotFoundError(f"No data found for the given time window: {start_datetime} to {end_datetime}")
```

Comment:
Why it's FileNotFoundError? if file doesn't exist, you will get an empty spark dataframe.

Reply:
change to DataNotFoundError and add a outer try-except to handle it.